src/s3fs.cpp

```diff
@@ -3368,10 +3368,16 @@ static int readdir_multi_head(const char* path, const S3ObjList& head, void* buf
 
         for(s3obj_list_t::iterator reiter = notfound_param.notfound_list.begin(); reiter != notfound_param.notfound_list.end(); ++reiter){
             int dir_result;
-            if(0 == (dir_result = directory_empty(reiter->c_str()))){
+            std::string reiter_str = *reiter;
```

Comment:
Your modification method probably correct, but I think it should be modified more.

- There were two problems:
One is that when calling `directory_empty()`, unless `dirpath`(=path + reiter_str) is used, it will not work if there are multiple layers of directories.
For the rest, I think `dir_result` should only be `-ENOTEMPTY`. (original code was wrong)

- After modifying these, I think the code will look like this:
```
std::string dirpath = path + (*reiter);
if(-ENOTEMPTY == (dir_result = directory_empty(dirpath.c_str()))){
    // Found objects under the path, so the path is directory.
    ...
```

Reply:
Hey @ggtakec , thanks for getting back to me! I am testing this approach with our use-case as we speak. Will update the code and rebase on `master` if it works :) I'll follow-up soon

Reply:
Verified that your solution works! Updated the code and rebased on `master`

Reply:
Thanks, I comfirmed it.